examples/meta/generator/translate.py

```diff
@@ -209,67 +209,45 @@ def dependenciesString(self, allClasses, interfacedClasses, enums):
             # Dependency strings are optional so we just return empty string
             return ""
 
-        # Three types of dependencies: a list of all classes used,
-        # a list of all explicitly constructed classes,
-        # and list of all enums used. All are optional.
-        allClassDependencies = ""
-        interfacedClassDependencies = ""
-        enumDependencies = ""
-        dependenciesExist = False
-
-        if len(allClasses) > 0 and "AllClassDependencies" in self.targetDict["Dependencies"]:
-            dependenciesExist = True
-            template = Template(self.targetDict["Dependencies"]["AllClassDependencies"])
-            allClassDependencies = template.substitute(classlist=self.seperatedClassDependencies(allClasses))
-
-        if len(interfacedClasses) > 0 and "InterfacedClassDependencies" in self.targetDict["Dependencies"]:
-            dependenciesExist = True
-            template = Template(self.targetDict["Dependencies"]["InterfacedClassDependencies"])
-            interfacedClassDependencies = template.substitute(classlist=self.seperatedClassDependencies(interfacedClasses))
-
-        if len(enums) > 0 and "EnumDependencies" in self.targetDict["Dependencies"]:
-            dependenciesExist = True
-            template = Template(self.targetDict["Dependencies"]["EnumDependencies"])
-            enumDependencies = template.substitute(enums=self.seperatedEnumDependencies(enums))
-
-        if not dependenciesExist:
-            return ""
-
-        allDependenciesTemplate = Template(self.targetDict["Dependencies"]["AllDependencies"])
-        return allDependenciesTemplate.substitute(allClassDependencies=allClassDependencies,
-                                                  interfacedClassDependencies=interfacedClassDependencies,
-                                                  enumDependencies=enumDependencies)
-
-    def seperatedClassDependencies(self, classes):
-        if len(classes) == 0:
-            return ""
+        dependencies = set()
 
-        dependencyList = list(classes)
+        if self.targetDict["Dependencies"].get("IncludeAllClasses"):
+            dependencies = dependencies.union(allClasses)
+        elif self.targetDict["Dependencies"].get("IncludeInterfacedClasses"):
+            dependencies = dependencies.union(interfacedClasses)
+        elif self.targetDict["Dependencies"].get("IncludeEnums"):
+            dependencies = dependencies.union(enums)
 
-        # Elements are just formatted as their names as default
-        elementTemplate = Template("$className")
-        if "DependencyListElementClass" in self.targetDict["Dependencies"]:
-            elementTemplate = Template(self.targetDict["Dependencies"]["DependencyListElementClass"])
+        translations = set(map(self.translateDependencyElement, dependencies))
```

Comment:
@karlnapf Here I'm making sure that each translated dependency is unique. I.e. if two different dependencies (an enum and a class) result in the same import translation, then they're squashed.